src/provider/provider_cuda.c

```diff
@@ -224,16 +263,21 @@ static umf_result_t cu_memory_provider_alloc(void *provider, size_t size,
         // this shouldn't happen as we check the memory_type settings during
         // the initialization
         LOG_ERR("unsupported USM memory type");
-        return UMF_RESULT_ERROR_UNKNOWN;
+        assert(false);
```

Comment:
Yeah, I agree. But I do not think we should set `cu_result` because it indicates a CUDA error (which is not true in that case). Since this situation is "not possible" as mentioned in the comment above I just `return UMF_RESULT_ERROR_UNKNOWN`

Reply:
makes sense, thx